### PR TITLE
[PLGN-245] Carbon Black Cloud - Added new action: Quarantine Multiple

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - Added new action: `Quarantine Multiple`

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/carbon_black_cloud/icon_carbon_black_cloud/util/utils.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                return True
+            else:
+                return False
+
+        # if 1.1.1.1/32 - remove /32
+        trimmed_address = re.sub(r"/32$", "", agent)
+
+        # if contains / we compare explicit matches, but not subnets in subnets
+        if "/" in trimmed_address:
+            return agent in whitelist
```
Comment (cmcnally-r7) [inline plugins/carbon_black_cloud/icon_carbon_black_cloud/util/utils.py:74]: Does this `if` ever get hit? It looks like if `/32` is removed, then `/` is never found in `trimmed_address`.

Are there cases where the ip has `/xx` that != `32`?

File: plugins/carbon_black_cloud/icon_carbon_black_cloud/actions/quarantine/action.py [hunk 2]
```diff
@@ -15,30 +15,11 @@ def __init__(self):
...
-            "options": {"toggle": toggle},
-        }
-
-        url = f"{self.connection.base_url}/appservices/v6/orgs/{self.connection.org_key}/device_actions"
-        self.connection.post_to_api(url, payload)
+        # END INPUT BINDING - DO NOT REMOVE
 
         # This API returns 204 no content if successful, we have to assume the state was applied on a successful call
-        return {Output.QUARANTINED: quarantine_state}
+        return {Output.QUARANTINED: self.connection.update_quarantine_state(agent, whitelist, quarantine_state)}
```
Comment (cmcnally-r7) [inline plugins/carbon_black_cloud/icon_carbon_black_cloud/actions/quarantine/action.py:25]: Would it be better to separate this logic just a little bit?
```suggestion
        response = self.connection.update_quarantine_state(agent, whitelist, quarantine_state)
        return {Output.QUARANTINED: response}
```

**Review by cmcnally-r7 — CHANGES_REQUESTED** [review]
lgtm just a couple of small queries 😄

File: plugins/carbon_black_cloud/icon_carbon_black_cloud/util/utils.py (hunk 1 above)
Comment (igorski-r7) [reply plugins/carbon_black_cloud/icon_carbon_black_cloud/util/utils.py:74]: It's about the subnet mask. It can be for example `/24` 😄

**Review by cmcnally-r7 — APPROVED** [review]

**Review by llaszuk-r7 — APPROVED** [review]

**Pull request merged**